gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/LocalRegion.java

```diff
@@ -14352,6 +14353,12 @@ private long getEntryOverhead(RegionEntry entry){
     Object key = entry.getRawKey();
     if (key != null) {
       entryOverhead += ReflectionSingleObjectSizer.INSTANCE.sizeof(key);
+    }else{
+      // first key.
+      Object firstKey = this.getRegionMap().keySet().iterator().next();
+      if(firstKey != null){
+        entryOverhead += ReflectionSingleObjectSizer.INSTANCE.sizeof(firstKey);
+      }
```

Comment:
Yes calcMemSize for key. But looking at it why is separate size accounting for key being done? Key overhead can change so its not a fixed overhead rather should be added for each entry separately.